csrc/device_lower/pass/inline_ptx.cpp

```diff
@@ -90,6 +90,72 @@ class LowerToInlinePtx : public kir::ExprMutator {
       return;
     }
   }
+
+  void handle(MmaOp* mma) override {
+    const int m = 16;
+    const int n = 8;
+    const int k = mma->isAmpere() ? 16 : 8;
+
+    std::string op;
+    {
+      std::stringstream op_ss;
+      op_ss << "mma.sync.aligned.m" << m << "n" << n << "k" << k
+            << ".row.col.f32";
+      if (mma->inA()->as<kir::TensorIndex>()->view()->getDataType().value() ==
+          DataType::BFloat16) {
+        op_ss << ".bf16.bf16";
+      } else {
+        op_ss << ".f16.f16";
+      }
+      op_ss << ".f32";
+      op = op_ss.str();
+    }
+
+    int64_t split_n = mma->n() / n;
+    int64_t split_k = mma->k() / k;
+
+    // If factor == 1, then do nothing, otherwise, view array<T, n> as
+    // array<array<T, n / factor>, factor>
+    auto maybe_outer_split = [](DataType dtype, int64_t factor) -> DataType {
+      if (factor == 1) {
+        return dtype;
+      }
+      const auto& array = std::get<ArrayType>(dtype.type);
+      return ArrayType{
+          std::make_shared<DataType>(
+              ArrayType{array.type, array.size / (size_t)factor}),
+          (size_t)factor};
+    };
+
+    DataType accumulator_type = maybe_outer_split(mma->out()->dtype(), split_n);
+    DataType a_type = maybe_outer_split(mma->inA()->dtype(), split_k);
+    DataType b_type = maybe_outer_split(mma->inB()->dtype(), split_n);
+    if (split_n > 1) {
+      auto& item_type = *std::get<ArrayType>(b_type.type).type;
+      item_type = maybe_outer_split(item_type, split_k);
```

Comment:
The result type (`item_type`) of this call will be:
```
array<
 array<array<T, n / split_n>,  split_n / split_k>,
 split_k
>
```
if `split_n` is greater than 1? I'm asking to be sure I understood correctly the result type generation and to be sure that I follow the access patterns generated in the nested loop at the bottom.

Reply:
Updated the PR to add a comment. This should be
```C++
array<array<array<T, n / split_n / split_k>, split_k>, split_n>
```